static/app/components/issueDiff/index.tsx

```diff
@@ -55,28 +57,69 @@ class IssueDiff extends Component<Props, State> {
   }
 
   fetchData() {
-    const {baseIssueId, targetIssueId, baseEventId, targetEventId} = this.props;
+    const {
+      baseIssueId,
+      targetIssueId,
+      baseEventId,
+      targetEventId,
+      organization,
+      project,
+    } = this.props;
+    const hasSimilarityEmbeddingsFeature = project.features.includes(
+      'similarity-embeddings'
+    );
 
     // Fetch component and event data
-    Promise.all([
-      import('../splitDiff'),
-      this.fetchEventData(baseIssueId, baseEventId ?? 'latest'),
-      this.fetchEventData(targetIssueId, targetEventId ?? 'latest'),
-    ])
-      .then(([{default: SplitDiffAsync}, baseEvent, targetEvent]) => {
+    const asyncFetch = async () => {
+      try {
+        const [baseEventData, targetEventData] = await Promise.all([
+          this.fetchEvent(baseIssueId, baseEventId ?? 'latest'),
+          this.fetchEvent(targetIssueId, targetEventId ?? 'latest'),
+        ]);
+
+        const [{default: SplitDiffAsync}, baseEvent, targetEvent] = await Promise.all([
+          import('../splitDiff'),
```

Comment:
lets declare the promise for `import('../splitDiff')` higher which will start the request.
```ts
const splitdiffPromise = import('../splitDiff');
```

and then you can just await it on its own line after, its kinda weird to mix regular requests and a component in a promise.all
```ts
const `{default: SplitDiffAsync} = splitdiffPromise;
```